src/Chainweb/Logging/Amberdata.hs

```diff
@@ -112,94 +127,219 @@ instance ToJSON AmberdataBlock where
+      ++ "If absent, will log Amberdata data from ALL chains.")
+
+
+pEnableAmberdataDebug
+  :: OptionParser Bool
+pEnableAmberdataDebug = enableDisableFlag
+  $ prefixLong (Just amberdataComponentPrefix) "debug"
+  <> suffixHelp Nothing "Whether to enable Amberdata debugging."
 
-instance ToJSON AmberdataConfig where
-    toJSON = String . amberdataConfigToText
 
-instance FromJSON AmberdataConfig where
-    parseJSON = parseJsonFromText "AmberdataConfig"
+data AmberdataConfig = AmberdataConfig
+    { _ambredataConfigHost :: HostAddress
+    , _amberdataApiKey :: AmberdataApiKey
+    , _amberdataBlockchainId :: AmberdataBlockchainId
+    , _amberdataChainId :: Maybe ChainId
+    , _amberdataDebug :: Bool
+    }
+    deriving (Show, Eq, Ord, Generic)
+
+makeLenses ''AmberdataConfig
+
+defaultAmberdataConfig :: AmberdataConfig
+defaultAmberdataConfig
+  = AmberdataConfig
+    defaultAmberdataHost
+    (AmberdataApiKey "")
+    (AmberdataBlockchainId "")
+    Nothing
+    True
+
+validateAmberdataConfig :: ConfigValidation AmberdataConfig []
+validateAmberdataConfig o = do
+  validateAmberdataHost $ _ambredataConfigHost o
+  validateAmberdataApiKey $ _amberdataApiKey o
+  validateAmberdataBlockchainId $ _amberdataBlockchainId o
+  validateOptionalChainId $ _amberdataChainId o
+
+pAmberdataConfig
+  :: MParser AmberdataConfig
+pAmberdataConfig = id
+  <$< ambredataConfigHost .:: pAmberdataHost
+  <*< amberdataApiKey .:: pAmberdataApiKey
+  <*< amberdataBlockchainId .:: pAmberdataBlockchainId
+  <*< amberdataChainId .:: pAmberdataOptionalChainId
+  <*< amberdataDebug .:: pEnableAmberdataDebug
+
+instance ToJSON AmberdataConfig where
+  toJSON o = object
+    [ "host" .= _ambredataConfigHost o
+    , "apiKey" .= _amberdataApiKey o
+    , "blockchainId" .= _amberdataBlockchainId o
+    , "chainId" .= _amberdataChainId o
+    , "debug" .= _amberdataDebug o
+    ]
+
+instance FromJSON (AmberdataConfig -> AmberdataConfig) where
+  parseJSON = withObject "AmberdataConfig" $ \o -> id
+    <$< ambredataConfigHost ..: "host" % o
+    <*< amberdataApiKey ..: "apiKey" % o
+    <*< amberdataBlockchainId ..: "blockchainId" % o
+    <*< amberdataChainId ..: "chainId" % o
+    <*< amberdataDebug ..: "debug" % o
 
 
 -- -------------------------------------------------------------------------- --
 -- Monitor
 
-amberdataBlockMonitor :: Logger logger => logger -> CutDb cas -> IO ()
-amberdataBlockMonitor logger db = do
+amberdataBlockMonitor :: (PayloadCas cas, Logger logger) => Maybe ChainId -> logger -> CutDb cas -> IO ()
+amberdataBlockMonitor cid logger db = do
     logFunctionText logger Info "Initialized Amberdata Block Monitor"
+    case cid of
+      Nothing -> logFunctionText logger Info "Sending blocks from ALL chains"
+      Just cid' -> logFunctionText logger Info ("Sending blocks from chain " <> toText cid')
     void
-        $ S.mapM_ (logAllBlocks logger)
-        $ S.map cutToAmberdataBlocks
+        $ S.mapM_ logBlocks
         $ cutStream db
```

Comment:
Consider using `blockStream` from `Chainweb.CutDB` instead of `CutStream`. It will be easier to use and give better results. In particular, it doesn't skip blocks and it logs each block only once (except for when chain jumps back and force between the same forks).